### PR TITLE
Implement details component

### LINKED ISSUE #349: Improved Input/Output variable mapping component is implemented and can be used in the properties panel
__Is your feature request related to a problem? Please describe__

We want to support the user when working with process variables. As part of this, the user shall be able to perform the input/output mapping more easily.

This feature covers the implementation of a new UI component which replaces the existing input/output mapping (which is implemented as list).

This component will then be used in the properties panel in multiple places
* General Input/Output tab for tasks
* Simplified property panel for tasks with applied element templates (see https://github.com/camunda/camunda-modeler/issues/1894)

__Describe the solution you'd like__

The current component looks like this:
Screenshot_20200818_120611

The target state of the new component can be found on [Zeplin](https://zpl.io/VKMepB7).

__Describe alternatives you've considered__

See [Zeplin](https://zpl.io/VKMepB7)

__Additional context__

Child of https://github.com/bpmn-io/internal-docs/issues/148
Required for https://github.com/camunda/camunda-modeler/issues/1894

### PR DESCRIPTION
**Pull request by barmac**
__Which issue does this PR address?__

Closes #349

__Acceptance Criteria__

* [ ] Corresponds to the concept 
* [ ] Corresponds to the design [Zeplin](https://zpl.io/VKMepB7)

__Definition of Done__

* [ ] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [ ] corresponds to [the code standards](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [ ] passes Continuous Integration checks
* [ ] available as feature branch on GitHub
  * [ ] contains the cleaned up commit history
  * [ ] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [ ] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

**Comment by barmac** [comment]
Ready for review.

**Comment by pinussilvestrus** [comment]
LOL

image

```html
<button class="bpp-collapsible__remove action-button clear" data-action="onRemove"></button>
```

does not equal

```html
<button class="bpp-collapsible__remove clear action-button" data-action="onRemove"></button>
```

will the class order always be the same or it is random? Just to prevent flaky tests.

**Comment by barmac** [comment]
Oh god I thought I fixed it...

**Comment by barmac** [comment]
I'll rewrite the tests so that they don't depend that much on html.

File: lib/factory/EntryFieldDescription.js [hunk 1]
```diff
@@ -8,9 +8,12 @@ var escapeHTML = require('../Utils').escapeHTML;
  * As a special feature, this description may contain both markdown,
  * plain <a href> links and <br />
  *
- * @param {String} description
+ * @param {string} description
+ * @param {object} [options]
+ * @param {string} [options.show] - name of callback to determine whether description is shown
  */
-module.exports = function entryFieldDescription(description) {
+module.exports = function entryFieldDescription(description, options) {
```
Comment (pinussilvestrus) [inline lib/factory/EntryFieldDescription.js:15]: 👍

**Review by pinussilvestrus — APPROVED** [review]
Besides the failing test case, it looks good to me 👏

**Pull request merged**